modelmapper/base.py

```diff
@@ -135,8 +134,7 @@ def _verify_no_duplicate_names(self, names):
             raise ValueError(f'The following fields were repeated in the csv: {duplicates}')
 
     def _get_clean_names_and_csv_data_gen(self, path):
-        delimiter = self.settings.csv_delimiter
```

Comment:
No need to read the delimiter from the settings since we always sniff it.

Reply:
Are you using the csv sniffer in python? Be careful. If so it has a line count requirement. I find it's really bad sniffing files lower than 500 or so lines.

Reply:
The Python sniffer sucks. I'm using `CleverCSV` sniffer. Are you sure it has a line count requirement? What I have seen is you can say how many bytes it should look at not lines.

Reply:
I think its bytes you're correct.